modules/apps/object/object-service/src/main/java/com/liferay/object/model/impl/ObjectRelationshipImpl.java

```diff
@@ -38,10 +39,19 @@ public boolean isEdgeCandidate() throws PortalException {
 			ObjectDefinitionLocalServiceUtil.getObjectDefinition(
 				getObjectDefinitionId1());
 
+		ObjectDefinition objectDefinition2 =
+			ObjectDefinitionLocalServiceUtil.getObjectDefinition(
+				getObjectDefinitionId2());
+
 		if (isSelf() ||
 			!Objects.equals(
-				ObjectRelationshipConstants.TYPE_ONE_TO_MANY, getType()) ||
-			objectDefinition1.isUnmodifiableSystemObject()) {
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY, getType())) {
+
+			return false;
+		}
```

Comment:
Hey @thallescezar here we'd prefer to declare objectDefinition1 and objectDefinition2 variables closer than their usages because it's make the code easier to read, so call the methods to get object definitions after this conditional, this way we'll also perform two fewer DB operations when the relationshiop is self of is not one to many.

Thank you!